tests/Unit/PointwiseFunctions/GeneralRelativity/Test_Christoffel.cpp

```diff
@@ -39,4 +61,81 @@ SPECTRE_TEST_CASE("Unit.PointwiseFunctions.GeneralRelativity.Christoffel.",
   test_christoffel<1, IndexType::Spacetime>(0.);
   test_christoffel<2, IndexType::Spacetime>(0.);
   test_christoffel<3, IndexType::Spacetime>(0.);
+
+  // Check that compute items work correctly in the DataBox
+  // First, check that the names are correct
+  CHECK(gr::Tags::SpatialChristoffelFirstKind<3, Frame::Inertial,
+                                              DataVector>::name() ==
+        "SpatialChristoffelFirstKind");
+  CHECK(gr::Tags::TraceSpatialChristoffelFirstKind<3, Frame::Inertial,
+                                                   DataVector>::name() ==
+        "TraceSpatialChristoffelFirstKind");
+  CHECK(gr::Tags::SpatialChristoffelSecondKind<3, Frame::Inertial,
+                                               DataVector>::name() ==
+        "SpatialChristoffelSecondKind");
+  CHECK(gr::Tags::TraceSpatialChristoffelSecondKind<3, Frame::Inertial,
+                                                    DataVector>::name() ==
+        "TraceSpatialChristoffelSecondKind");
+
+  CHECK(gr::Tags::SpatialChristoffelFirstKindCompute<3, Frame::Inertial,
+                                                     DataVector>::name() ==
+        "SpatialChristoffelFirstKind");
+  CHECK(gr::Tags::TraceSpatialChristoffelFirstKindCompute<3, Frame::Inertial,
+                                                          DataVector>::name() ==
+        "TraceSpatialChristoffelFirstKind");
+  CHECK(gr::Tags::SpatialChristoffelSecondKindCompute<3, Frame::Inertial,
+                                                      DataVector>::name() ==
+        "SpatialChristoffelSecondKind");
+
+  // Check that the compute items return correct values
+  const DataVector test_vector{5., 4.};
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> distribution(-0.2, 0.2);
+
+  auto spacetime_metric =
+      make_with_random_values<tnsr::aa<DataVector, 3, Frame::Inertial>>(
+          make_not_null(&generator), make_not_null(&distribution), test_vector);
```

Comment:
is the `test_vector` here just used for a "used_for_size" argument? If so, I think that it should now work to just use the size in the argument (new overloads have been added such that if you just provide the numeric size it should work as expected).

Reply:
If instead of `test_vector`, I provide a `size_t` for size, I get compile time warning:

```
/home/prayush/src/spectre/spectre_3p1derivs_01/tests/Unit/PointwiseFunctions/GeneralRelativity/Test_Christoffel.cpp:11:
In file included from /home/prayush/src/spectre/spectre_3p1derivs_01/src/DataStructures/DataBox/DataBox.hpp:17:
In file included from /home/prayush/src/spectre/spectre_3p1derivs_01/src/DataStructures/DataBox/DataBoxTag.hpp:15:
In file included from /home/prayush/src/spectre/spectre_3p1derivs_01/src/Utilities/ConstantExpressions.hpp:20:
/home/prayush/src/spectre/spectre_3p1derivs_01/src/Utilities/MakeWithValue.hpp:22:34: warning: inline function
      'MakeWithValueImpls::MakeWithValueImpl<Tensor<DataVector, brigand::list<brigand::integral_constant<int, 1>,
      brigand::integral_constant<int, 1> >, brigand::list<Tensor_detail::TensorIndexType<3, UpLo::Lo, Frame::Inertial,
      IndexType::Spacetime>, Tensor_detail::TensorIndexType<3, UpLo::Lo, Frame::Inertial, IndexType::Spacetime> > >, unsigned
      long>::apply<double>' is not defined [-Wundefined-inline]
  static SPECTRE_ALWAYS_INLINE R apply(const T& input,
```

Then I looked for use cases of the overload you referred to, and didn't find any in `tests/Unit/Utilities/Test_MakeWithValue.cpp`. Could you please point me to the overload of `make_with_random_values` that takes in the size directly?

Reply:
Ah, I see. It looks like the trouble is that the MakeWithValues for tensor isn't forwarding as I expected to the vector version for its components. 
The main use case I know about (and why I wrote it) is in `VectorImplTestHelper.hpp`, line 510, though I fear the nature of those test helpers makes it a bit difficult tor read, and that only used the trick for vectors.
Given that, I'm happy with leaving this as it is, and I'll make an issue for myself to make it work for tensors in the future.

Reply:
Hmmm, makes sense. Although, even in `VectorImplTestHelper.hpp`,  a `used_for_size` is passed to `make_with_random_values` at several places e.g. line 179, 207, 277.. 

In any case, I have fixed and rebased this, as well as  #1476 and #1477.

Reply:
Oh, you're right. I guess I never went back and simplified the rest of those to avoid constructing the extra vector for most of those calls. I'll add that to the issue :)